src/api.js

```diff
@@ -0,0 +1,59 @@
+
+const Api = /** @class */ (function() {
```

Comment:
I could be persuaded on a different name for this. This class is essentially helper functions that are disposable (i.e. any timers, listeners, etc. that they create can be torn down when a key becomes inactive).

There's only one helper function now, but my thought was this could form the basis for library functions supplied to variant functions that we have been discussing providing for ES. If this idea is YAGNI, then I can slim down the code to a more minimal disposer mechanism.

Reply:
We're not thinking of building out a large library of helper functions -- intending to use integrations for that. So I'd lean simpler

Reply:
I'll pare it down. I will note, though, that operations like this that need disposal can't be implemented via Integrations at the moment. We could potentially allow async variant functions to return, in addition to `true`, a disposer function that is called on disposal.